### PR TITLE
Feat: Handle both addDomain and editDomain actions created via a Multi-Sig decision method

### PR DESCRIPTION
**Pull request by rumzledz**
This is needed to test [colonyCDapp#2588: Feat: enable addDomain multi-sig flow](https://github.com/JoinColony/colonyCDapp/pull/2588)

### CODE REVIEW

**Review by bassgeta — CHANGES_REQUESTED** [review]
Codewise it all looks good, we can get rid of that TODO comment too.
Good job, I've left some comments on the CDapp PR which should be resolved before we get this one in.

File: src/handlers/multiSig/multiSigExecuted/multiSigExecuted.ts [hunk 1]
```diff
@@ -35,6 +37,29 @@ export const handleMultiSigMotionExecuted: EventHandler = async (
...
+    }
+
+    await linkPendingMetadata(
+      actionData,
+      colonyAddress,
+      finalizedMultiSig.id,
+      true,
+    );
+
     /* @TODO fix this up when we start porting over domain motions
```
Comment (iamsamgibbs) [inline src/handlers/multiSig/multiSigExecuted/multiSigExecuted.ts:63]: This comment can go now that you've implemented it

File: src/handlers/multiSig/multiSigCreated/handlers/addOrEditDomain.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  const { name } = parsedAction;
+
+  const pendingDomainMetadataId = getPendingMetadataDatabaseId(
+    colonyAddress,
+    event.transactionHash,
+  );
+
+  await createMultiSigInDB(colonyAddress, event, {
+    type: multiSigNameMapping[name],
+    pendingDomainMetadataId,
```
Comment (iamsamgibbs) [inline src/handlers/multiSig/multiSigCreated/handlers/addOrEditDomain.ts:24]: We might need to add this to show the `CreatedIn` field on the completed action screen:

```ts
const { name, args } = parsedAction;
const [, , nativeDomainId] = args;

...

fromDomainId: getDomainDatabaseId(colonyAddress, nativeDomainId),
```

(I could be wrong, but it is there for the `editDomain` reputation motion)

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Looks good to me and is working as intended. Just one comment to remove, and one little bit to investigate whilst you fix up the CDapp stuff!

File: src/handlers/multiSig/multiSigCreated/handlers/addOrEditDomain.ts (hunk 2 above)
Comment (rumzledz) [reply src/handlers/multiSig/multiSigCreated/handlers/addOrEditDomain.ts:24]: Hey @iamsamgibbs ! At first I tried to do that before I realised that Kerry has already "pre-added" this inside the `multiSigCreated/helpers.ts` file

Screenshot 2024-07-03 at 14 37 44

File: src/handlers/multiSig/multiSigExecuted/multiSigExecuted.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (rumzledz) [reply src/handlers/multiSig/multiSigExecuted/multiSigExecuted.ts:63]: Aye!

Comment (rumzledz) [reply src/handlers/multiSig/multiSigExecuted/multiSigExecuted.ts:63]: Aye!

**Review by bassgeta — APPROVED** [review]
Tested the CDapp and everything works, so let's get this merged!

**Review by mmioana — APPROVED** [review]
Awesome @rumzledz 👏 

Even if the block ingestor doesn't treat createDomain and editDomain differently, would it make sense to have two separate handlers?

**Comment by rumzledz** [comment]
> Awesome @rumzledz 👏
> 
> Even if the block ingestor doesn't treat createDomain and editDomain differently, would it make sense to have two separate handlers?

Hey @mmioana it does at first glance but after a dozen conversations about this whole code split thing, it's not worth it. The cdapp case is different and so I did the code split there 👌

**Review by iamsamgibbs — APPROVED** [review]

**Review by bassgeta — APPROVED** [review]
Reapproving since reviewed as part of its CDapp counterpart :+1:

**Review by rdig — APPROVED** [review]
Tested as part of JoinColony/colonyCDapp#2588

Nicely done once again

**Pull request merged**